terraform/aws-https/infra/infra.tf

```diff
@@ -0,0 +1,724 @@
+      port        = "443"
+      protocol    = "HTTPS"
+      status_code = "HTTP_301"
+    }
+  }
+}
+
+# ALB default HTTPS backend
+resource "aws_lb_listener" "wandb_https_listener" {
+  load_balancer_arn = aws_lb.wandb.arn
+  port              = 443
+  protocol          = "HTTPS"
+  ssl_policy        = "ELBSecurityPolicy-2016-08"
+  certificate_arn   = aws_acm_certificate_validation.costah_dev.certificate_arn
+
+  default_action {
+    type = "forward"
+    target_group_arn = aws_lb_target_group.wandb_tg.arn
+  }
+}
+
+resource "aws_autoscaling_attachment" "wandb" {
+  autoscaling_group_name = aws_eks_node_group.eks_worker_node_group.resources[0].autoscaling_groups[0].name
+  alb_target_group_arn   = aws_lb_target_group.wandb_tg.arn
+}
+
+##########################################
+# SQS/SNS
+##########################################
+
+resource "aws_sqs_queue" "file_metadata" {
+  name = "wandb-file-metadata"
+
+  # enable long-polling
+  receive_wait_time_seconds = 10
+}
+
+output "sqs_queue_name" {
+  value = aws_sqs_queue.file_metadata.name
+}
+
+resource "aws_sqs_queue_policy" "file_metadata_queue_policy" {
+  queue_url = aws_sqs_queue.file_metadata.id
+
+  policy = data.aws_iam_policy_document.file_metadata_queue_policy.json
+}
+
+data "aws_iam_policy_document" "file_metadata_queue_policy" {
+  statement {
+    actions   = ["SQS:SendMessage"]
+    effect    = "Allow"
+    resources = [aws_sqs_queue.file_metadata.arn]
+
+    principals {
+      type        = "Service"
+      identifiers = ["sns.amazonaws.com"]
+    }
+
+    condition {
+      test     = "ArnLike"
+      variable = "aws:SourceArn"
+      values   = [aws_sns_topic.file_metadata.arn]
+    }
+  }
+}
+
+resource "aws_sns_topic" "file_metadata" {
+  name = "wandb-file-metadata-topic"
+}
+
+resource "aws_sns_topic_policy" "file_metadata_topic_policy" {
+  arn = aws_sns_topic.file_metadata.arn
+
+  policy = data.aws_iam_policy_document.file_metadata_topic_policy.json
+}
+
+data "aws_iam_policy_document" "file_metadata_topic_policy" {
+  statement {
+    sid       = "s3-can-publish"
+    actions   = ["SNS:Publish"]
+    effect    = "Allow"
+    resources = [aws_sns_topic.file_metadata.arn]
+
+    principals {
+      type        = "Service"
+      identifiers = ["s3.amazonaws.com"]
+    }
+
+    condition {
+      test     = "ArnLike"
+      variable = "aws:SourceArn"
+      values   = [aws_s3_bucket.file_storage.arn]
+    }
+  }
+}
+
+resource "aws_sns_topic_subscription" "file_metadata" {
+  topic_arn = aws_sns_topic.file_metadata.arn
+  protocol  = "sqs"
+  endpoint  = aws_sqs_queue.file_metadata.arn
+}
+
+##########################################
+# S3
+##########################################
+
+resource "aws_s3_bucket" "file_storage" {
+  bucket        = "${var.global_environment_name}-wandb-files"
+  acl           = "private"
+  force_destroy = true
+
+  cors_rule {
+    allowed_headers = ["*"]
+    allowed_methods = ["GET", "HEAD", "PUT"]
+    allowed_origins = ["*"]
```

Comment:
For security sake I would rather use something more restricted here, like `"*.${module.infra.lb_dns_name}"` or see `AllowedOrigin` at https://docs.wandb.ai/guides/self-hosted/configuration

Reply:
Because this is a custom domain setup, I think it should use the `wandb.${aws_route53_zone.costah_dev.name}`?